tlv/complex.go

```diff
@@ -0,0 +1,42 @@
+package tlv
+
+import (
+	"io"
+
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+// EKeyLocator is an encoder for keychain.KeyLocator.
```

Comment:
That makes sense. There's a larger discussion I think that should be had about standalone libraries in the `lnd` repo because then they can't follow the mocking pattern in `lntest/mock` due to import cycles. Though that may be fixable by properly splitting up `lntest/mock`.

Reply:
what sort of import cycles? i'm assuming you mean import cycles in test frameworks? this can be easily avoided by writing all tests for `<pkg>` in `<pkg>_test`.

an example of this is `wtdb`, where `wtmock` relies on types in `wtdb`. the test is in the same directory but uses the `wtdb_test` package, which a) is a work around for the import cycle, and b) ensures that the package's public api is tested, and isn't relying on internal members/functions/variables

Reply:
I guess it comes down to package-level mocking like `watchtower/wtmock` or the way I structured it in `lntest/mock`. `lntest/mock` will have more import cycles because everything isn't properly separated by package which is something I probably should have caught earlier.

Reply:
`channeldb` or `keychain`?

Reply:
i'd do `channeldb` since that's where we keep all of our other serialization, and also the TLV type definition